### PR TITLE
Update signedAssertion test to avoid resource contention

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs [hunk 1]
```diff
@@ -3,81 +3,76 @@
...
 {
     public class SignedAssertionFilePathCredentialsLoaderTests
     {
-        const string filePath = "signedAssertion.txt";
-        const string aksEnvironmentVariableName = "AZURE_FEDERATED_TOKEN_FILE";
-        string token;
-        SignedAssertionFilePathCredentialsLoader signedAssertionFilePathCredentialsLoader = new SignedAssertionFilePathCredentialsLoader(null);
-
+        private const string FilePath = "signedAssertion.txt";
+        private const string FilePath2 = "signedAssertion2.txt";
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs:16]: that's clever

File: tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs [hunk 2]
```diff
@@ -3,81 +3,76 @@
...
+        private const string AksEnvironmentVariableName = "AZURE_FEDERATED_TOKEN_FILE";
+        private readonly string _token;
+        private readonly SignedAssertionFilePathCredentialsLoader _signedAssertionFilePathCredentialsLoader = new(null);
 
         public SignedAssertionFilePathCredentialsLoaderTests()
         {
-            JsonWebTokenHandler handler = new JsonWebTokenHandler();
-            token = handler.CreateToken("{}");
+            JsonWebTokenHandler handler = new();
+            _token = handler.CreateToken("{}");
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs:24]: would we want to write the files too here?

File: tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs [hunk 3]
```diff
@@ -3,81 +3,76 @@
...
-        }
-
-        [Fact]
-        public async Task GetClientAssertion_WhenEnvironmentVariablePointsToSignedAssertionFileExists_ReturnsClientAssertion()
-        {
-            // Arrange
-            File.WriteAllText(filePath, token.ToString());
-            Environment.SetEnvironmentVariable(aksEnvironmentVariableName, filePath);
-            CredentialDescription credentialDescription = new CredentialDescription
+            if (withEnvVariable)
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs:61]: does it build? It seem that this is code outside of a method? (from what the diff provides in GitHub. Might be a tooling issue?

**Review by jmprieur — APPROVED** [review]
LGTM

File: tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs (hunk 3 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs:61]: I think it's a tooling issue as i've been running the tests locally.

File: tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs (hunk 2 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Web.Test/SignedAssertionFilePathCredentialsLoader.cs:24]: I dn't think it's necessary.

File: tests/Microsoft.Identity.Web.Test/AzureIdentityForKubernetesClientAssertionTests.cs [hunk 4]
```diff
@@ -25,25 +27,22 @@ public AzureIdentityForKubernetesClientAssertionTests()
         public async Task GetAksClientAssertion_WhenSpecifiedSignedAssertionFileExists_ReturnsClientAssertion()
         {
             // Arrange
-            File.WriteAllText(TestConstants.signedAssertionFilePath, token.ToString());
+            File.WriteAllText(FilePath, token.ToString());
             AzureIdentityForKubernetesClientAssertion azureIdentityForKubernetesClientAssertion = new AzureIdentityForKubernetesClientAssertion(TestConstants.signedAssertionFilePath);
```
Comment (jmprieur) [inline tests/Microsoft.Identity.Web.Test/AzureIdentityForKubernetesClientAssertionTests.cs:31]:
```suggestion
            AzureIdentityForKubernetesClientAssertion azureIdentityForKubernetesClientAssertion = new AzureIdentityForKubernetesClientAssertion(FilePath);
```

**Pull request merged**